doc/adrs/095_simplify_the_manipulation_of_nodes_by_switching_to_a_common_box_model.adoc

```diff
@@ -0,0 +1,458 @@
+
+The following nodes all have the same size:
+
+image:box_model_node_size.png[Various nodes with the same size]
+
+===== Size and contraints
+
+Nodes will have multiple types of sizes, such as:
+
+- a current size
+- a prefered size
+- a minimal size
+
+The current size is, of course, the size affected to a node at a given time.
+It may be null if the size of the node has not been computed yet.
+
+The prefered size of a node is the size of the node without any constraint.
+It can be configured by a specifier or by default in the application.
+You can think of it as the size the node should take if its parent has a free form layout strategy.
+For example, one could expect all its empty nodes associated with a specific node description to have a size of 150x70.
+
+The minimal size is the smallest size a node can take at any moment.
+It is directly influenced by the size of its content and its child layout strategy.
+As an example, image that the node described before with a prefered size of 150x70 has a minimal size of 10x10.
+On the other hand, let's consider that this node now has a large label, multiple children and large number of border nodes.
+This forces it to have a minimal size of 250x120.
+
+
+==== Footprint
+
+The following boxes will count torward the footprint of the node, specified also using a `width` and a `height`, when layouted in a parent but they will not be part of the size of the node.
+
+- border nodes box
+- outside label box
+- margin box
+
+As a result, our node of 100x100 could have a footprint of 200x150 because of its border nodes, outside label and margin.
+We would thus have to consider this footprint of 200x150 in order to find out how this node would overlap its siblings.
+In the example below, we can see various nodes of the same size but various footprints.
+
+image:box_model_node_footprint.png[Various nodes with the same size but different footprints]
+
+The child layouting strategy of the parent may impose a specific footprint for our node.
+The requested footprint will always have to be at least the minimal footprint of our node.
+If the requested footprint is bigger than our current footprint, then we will have to ask the child to render itself with a bigger footprint.
+For example, if a child node is located in a large parent node with a list layout, this will not change the size of the border of the child node.
+
+
+==== Footprint and constraints
+
+In some use case, to match a footprint requested by our parent, we will have to compute another footprint for a node.
+The configuration of a node should indicate which boxes should be resized in order to match the newly requested footprint.
+
+For example, if we consider a node with a list layout in a column, it may ask some of its children to occupy a larger footprint than necessary.
+In one case, the specifier may want those children to expand their outside label box to occupy this larger footprint, for example for the attributes of a class diagram.
```

Comment:
I don't understand why you are referring outside label box here, but in the associated picture below, it seems inside label boxes are used.